drivers/modem/modem_socket.c

```diff
@@ -368,6 +374,24 @@ void modem_socket_data_ready(struct modem_socket_config *cfg,
 	k_sem_give(&cfg->sem_lock);
 }
 
+int modem_socket_flush(struct modem_socket_config *cfg)
```

Comment:
I don't buy this function name, so please try to convince me it is correct. `flush` makes me think that this function will make sure all outgoing packets will be actually sent, at least offloaded to chip that implements network stack. However this implementation closes all sockets just like that. When it comes to `flush` in function name, I expect it to be "compatible" with `fflush()` (`man 3 fflush`) behavior.

Reply:
@mniestroj You are right, this is not a flush, it's more like a "make sure there is no remaining socket after device power off".
So in fact an hack for catching potential socket leak.
I therefore think I should simply remove the corresponding commit from the PR.